x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/30_semantic_text_inference.yml

```diff
@@ -0,0 +1,551 @@
+  - match: {test-dense-index.mappings.properties.another_inference_field.type: semantic_text}
+  - match: {test-dense-index.mappings.properties.non_inference_field.type: text}
+  - length: {test-dense-index.mappings.properties: 3}
+
+---
+"Sparse vector results are indexed as nested chunks and searchable":
+  - do:
+      bulk:
+        index: test-sparse-index
+        refresh: true
+        body: |
+          {"index":{}}
+          {"inference_field": ["you know, for testing", "now with chunks"]}
+          {"index":{}}
+          {"inference_field": ["some more tests", "that include chunks"]}
+
+  - do:
+      search:
+        index: test-sparse-index
+        body:
+          query:
+            nested:
+              path: inference_field.inference.chunks
+              query:
+                text_expansion:
+                  inference_field.inference.chunks.embeddings:
+                    model_id: sparse-inference-id
+                    model_text: "you know, for testing"
+
+  - match: { hits.total.value: 2 }
+  - match: { hits.total.relation: eq }
+
+  # Search with inner hits
+  - do:
+      search:
+        _source: false
+        index: test-sparse-index
+        body:
+          query:
+            nested:
+              path: inference_field.inference.chunks
+              inner_hits:
+                _source: false
+                fields: [inference_field.inference.chunks.text]
+              query:
+                text_expansion:
+                  inference_field.inference.chunks.embeddings:
+                    model_id: sparse-inference-id
+                    model_text: "you know, for testing"
+
+  - match: { hits.total.value: 2 }
+  - match: { hits.total.relation: eq }
+  - match: { hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.total.value: 2 }
+  - exists:  hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.hits.0.fields.inference_field\.inference\.chunks.0.text
+  - exists:  hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.hits.1.fields.inference_field\.inference\.chunks.0.text
+
+
+---
+"Dense vector results are indexed as nested chunks and searchable":
+  - do:
+      bulk:
+        index: test-dense-index
+        refresh: true
+        body: |
+          {"index":{}}
+          {"inference_field": ["you know, for testing", "now with chunks"]}
+          {"index":{}}
+          {"inference_field": ["some more tests", "that include chunks"]}
+
+  - do:
+      search:
+        index: test-dense-index
+        body:
+          query:
+            nested:
+              path: inference_field.inference.chunks
+              query:
+                knn:
+                  field: inference_field.inference.chunks.embeddings
+                  query_vector_builder:
+                    text_embedding:
+                      model_id: dense-inference-id
+                      model_text: "you know, for testing"
+
```

Comment:
No assertions after this search?

Reply:
there are some assertions on the total hits returned, and I added checks for the chunks as well.